pkg/util/timeutil/pgdate/pgdate.go

```diff
@@ -107,13 +108,14 @@ func MakeDateFromTime(t time.Time) (Date, error) {
 	return d, err
 }
 
+var errDateOutOfRange = errors.WithCandidateCode(errors.New("date is out of range"), pgcode.DatetimeFieldOverflow)
+
 // MakeDateFromUnixEpoch creates a Date from the number of days since the
 // Unix epoch.
 func MakeDateFromUnixEpoch(days int64) (Date, error) {
 	days, ok := arith.SubWithOverflow(days, unixEpochToPGEpoch)
 	if !ok || days <= math.MinInt32 || days >= math.MaxInt32 {
-		return Date{}, pgerror.New(pgerror.CodeDatetimeFieldOverflowError,
-			"date is out of range")
+		return Date{}, errDateOutOfRange
```

Comment:
Do we care that this breaks stack printing? That is, the error's stack is not from where it happened, but from where the variable was init'd.